
https://www.baeldung.com/wildfly-ejb-jndi






systemctl restart mysqld

https://www.baeldung.com/wildfly-ejb-jndi

Caused by: javax.security.sasl.SaslException: Authentication failed: all available authentication mechanisms failed:
   DIGEST-MD5: javax.security.sasl.SaslException: DIGEST-MD5: Server rejected authentication

env.put(Context.PROVIDER_URL, "http-remoting://localhost:8080");
    env.put(Context.INITIAL_CONTEXT_FACTORY, "org.jboss.naming.remote.client.InitialContextFactory");
    env.put(Context.SECURITY_PRINCIPAL, "client");
    env.put(Context.SECURITY_CREDENTIALS, "q");

    Context ctx = new InitialContext(env);
    connectionFactory = (ConnectionFactory)ctx.lookup("/jms/RemoteConnectionFactory");
    connection = connectionFactory.createConnection("client", "q");


I fixed the error with below changes.
It looks like i was using ApplicationRealm for remote authentication as well. I updated below line with my jaas realm. 
<http-connector name="http-remoting-connector" connector-ref="default" security-realm="ApplicationRealm"/>


1. Adding "username="jboss" security-realm="ManagementRealm" to host.xml (my host controller) which was previously missing in my configuration which I believe forces the host controller to validate using the username "jboss" 
 
          <remote host="${jboss.domain.master.address}" port="${jboss.domain.master.port:9999}" username="jboss" security-realm="ManagementRealm"/>
 
2. And then as mentioned in this post I added the user "jboss" to the domain controller.
 
3. Then I added Base64 encoded password which was generated when I created the user "jboss" to the host controller. This looks something like the following:
 
     <server-identities>
                    <!-- Replace this with either a base64 password of your own, or use a vault with a vault expression -->
         <secret value="aG9zdDFwYXNzd29yZA=="/>
     </server-identities>

   <security-realm name="ejb-security-realm">
    <server-identities>
      <secret value="cf86fc49735679542aa078d3d4a976be"/>
    </server-identities>
    </security-realm>
...
    <subsystem xmlns="urn:jboss:domain:remoting:1.1">
      <connector name="remoting-connector" socket-binding="remoting" security-realm="ApplicationRealm" />
        <outbound-connections>
    <remote-outbound-connection name="remote-ejb-connection" outbound-socket-binding-ref="remote-ejb" security-realm="ejb-security-realm" username="user">
      <properties>
        <property name="SASL_POLICY_NOANONYMOUS" value="false"/>
        <property name="SSL_ENABLED" value="false"/>
      </properties>
    </remote-outbound-connection>
  </outbound-connections>
...
  <outbound-socket-binding name="remote-ejb">
    <remote-destination host="host1" port="4447"/>
  </outbound-socket-binding>


Also, i set SASL_POLICY_NOANONYMOUS and SASL_POLICY_NOPLAINTEXT to false on client side. 


http://middlewaremagic.com/jboss/?tag=security


SASL_POLICY_NOANONYMOUS and SASL_POLICY_NOPLAINTEXT to false on client side. 



Root exception is javax.security.sasl.SaslException: Authentication failed: all available authentication mechanisms failed:
https://python5.com/q/dyrznuix